From 37057a9fcbea1858477134c70920d4577ebadf56 Mon Sep 17 00:00:00 2001
From: Xinhui Yang <cyan@cyano.uk>
Date: Mon, 22 Jun 2026 18:59:06 +0800
Subject: [PATCH 7/7] AOSCOS: Add AOSC-specific ARMv4, v6 and v7 definitions

---
 data/cputable   | 5 ++++-
 data/tupletable | 3 +++
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/data/cputable b/data/cputable
index 1a34b9869..dfc6bff0d 100644
--- a/data/cputable
+++ b/data/cputable
@@ -22,7 +22,10 @@ alpha		alpha		alpha.*			64	little
 amd64		x86_64		(amd64|x86_64)		64	little
 arc		arc		arc			32	little
 armeb		armeb		arm.*b			32	big
-arm		arm		arm.*			32	little
+armv4l		armv4l		armv4.*l		32	little
+armv6lhf	armv6l		armv6.*l		32	little
+armv7lhf	armv7l		armv7.*l		32	little
+arm		arm		arm[^467]*		32	little
 arm64		aarch64		aarch64			64	little
 hppa		hppa		hppa.*			32	big
 loongarch64	loongarch64	loongarch64		64	little
diff --git a/data/tupletable b/data/tupletable
index e4f4b53ab..26e2fc7bf 100644
--- a/data/tupletable
+++ b/data/tupletable
@@ -25,7 +25,10 @@ eabi-uclibc-linux-arm		uclibc-linux-armel
 base-uclibc-linux-<cpu>		uclibc-linux-<cpu>
 eabihf-musl-linux-arm		musl-linux-armhf
 base-musl-linux-<cpu>		musl-linux-<cpu>
+eabihf-gnu-linux-armv6lhf	armv6hf
+eabihf-gnu-linux-armv7lhf	armv7hf
 eabihf-gnu-linux-arm		armhf
+eabi-gnu-linux-armv4l		armv4
 eabi-gnu-linux-arm		armel
 abin32-gnu-linux-mips64r6el	mipsn32r6el
 abin32-gnu-linux-mips64r6	mipsn32r6
-- 
2.52.0

